src/App.js

```diff
@@ -1,58 +1,45 @@
-import React from 'react'
+import React, { useState } from 'react'
+
 import FakeSearchAPI from './fake-api/search'
 import { PAutoComplete } from './components/widgets'
+import { debounce } from './utils'
 import './App.css'
 
-class App extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      keyword: '',
-      results: [],
-    }
-  }
-  updateKeyword(keyword) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          keyword,
-        },
-        () => resolve(keyword)
-      )
-    })
-  }
-  updateResult(results) {
-    return new Promise((resolve) => {
-      this.setState(
-        {
-          results,
-        },
-        () => resolve()
-      )
-    })
-  }
-  async search(keyword) {
+const App = () => {
+  const [loading, setLoading] = useState(false)
+  const [suggestions, setSuggestions] = useState([])
+
+  const search = debounce(async (keyword) => {
+    if(!keyword) return;
```

Comment:
By ignoring `empty` keyword, this make inconsistent state between inner suggestions and outer suggestions. Reproduce steps:
> Given input is empty
> Then both inner and outer suggestions are empty
> When type 'b' in the input
> Then both inner and outer suggestion are ["Bethany Holmes"]
> When remove 'b' from the input
> The inner suggestions change to empty list but the outer suggestion still remain ["Bethany Holmes"]
This can lead to potential bugs.